Assets/Scripts/Gameplay/Player/PlayerMovement.cs

```diff
@@ -8,21 +8,33 @@ public class PlayerMovement : MonoBehaviour
 
     float horizontal;
     float vertical;
-
+    private PlayerStats playerStats;
     public float speed = 5.0f;
 
 
     // Start is called before the first frame update
     void Start()
     {
         body = GetComponent<Rigidbody2D>();
+        playerStats = GetComponent<PlayerStats>();
```

Comment:
same thing for these two, Add a requiresComponent tag to the class above if you are going to assume it has them or null check them. The requirescomponents makes it so that if you add this script to an object that doesn't have a PlayerStats script or a RigidBody it will automatically add them on.